src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/letterdetails/managehearings/VacateHearingNoticeLetterDetailsMapper.java

```diff
@@ -24,24 +28,35 @@ public VacateHearingNoticeLetterDetailsMapper(CourtDetailsConfiguration courtDet
     @Override
     public DocumentTemplateDetails buildDocumentTemplateDetails(FinremCaseDetails caseDetails) {
         FinremCaseData caseData = caseDetails.getData();
-        Hearing hearing = getWorkingHearing(caseData);
+
+        ManageHearingsWrapper hearingsWrapper =  caseData.getManageHearingsWrapper();
+        WorkingVacatedHearing workingVacatedHearing = hearingsWrapper.getWorkingVacatedHearing();
+        UUID workingVacatedHearingId = hearingsWrapper.getWorkingVacatedHearingId(workingVacatedHearing);
+
+        VacateOrAdjournedHearing vacatedOrAdjournedHearing =
+            hearingsWrapper.getVacatedOrAdjournedHearingsCollectionItemById(workingVacatedHearingId).getValue();
```

Comment:
NPE potentially

Reply:
Intentional.  If there is an NPE, then I want this to stop.  I don't want to potentially generate a document that's incorrect (caseworker may miss it).

If this was likely to happen, an exception feels more appropriate, will add one.